File: Verse.java
public class Verse {
    String data;
    String source;

    public Verse(String str, String book) {
        String tempSourceStr = "";
        int endIndex = str.indexOf(" ");
        if (book.equals("שמואל") || book.equals("מלכים") || book.equals("דברי הימים") || book.equals("עזרא / נחמיה")) {
            int firstSpacePos = str.indexOf(' ');
            endIndex = str.indexOf(' ', firstSpacePos + 1);
        }
        tempSourceStr = book + " " + str.substring(0, endIndex);
        this.source = tempSourceStr.replaceAll("[\\s\\n\\t-]", " ").replaceAll("\\s+", " ");
        this.data = str.substring(str.indexOf(" ")).replaceAll("[,;:]", "").replaceAll("[\\s\\n\\t-]", " ").replaceAll("\\s+", " ");
    }

    public String getData() {
        return data;
    }

    public String getSource() {
        return source;
    }

    public char getFirstLetter() {
        int index = 0;
        while (Integer.toHexString(getData().charAt(index)).equals("20")) {
            index++;
        }
        return getData().charAt(index);
    }

    public char getLastLetter() {
        String data = getData();
        return data.charAt(data.length() - 1);
    }
}

File: Verse.cs
namespace ConsoleApplication1
{
    public class Verse
    {
        public string Data { get; }
        public string Source { get; }

        public Verse(string str, string book)
        {
            var endIndex = str.IndexOf(" ");
            if (book == "שמואל" || book == "מלכים" || book == "דברי הימים" || book == "עזרא / נחמיה")
            {
                var firstSpacePos = str.IndexOf(' ');
                endIndex = str.IndexOf(' ', firstSpacePos + 1);
            }
            var tempSourceStr = book + " " + str.Substring(0, endIndex);
            Source = System.Text.RegularExpressions.Regex.Replace(tempSourceStr, @"[\s\n\t-]", " ").Replace(" ", string.Empty);
            Data = System.Text.RegularExpressions.Regex.Replace(str.Substring(str.IndexOf(" ")), "[,;:]", "").Replace(" ", string.Empty);
        }

        public char GetFirstLetter()
        {
            var index = 0;
            while (char.ConvertToUtf32(Data, index).ToString("X4") == "0020")
            {
                index++;
            }
            return Data[index];
        }

        public char GetLastLetter()
        {
            return Data[Data.Length - 1];
        }
    }
}
